packages/braavos-connector/src/index.ts

```diff
@@ -0,0 +1,110 @@
+import { AbstractConnector } from '@web3-starknet-react/abstract-connector';
+import { ConnectorUpdate } from '@web3-starknet-react/types';
+import { AccountInterface } from 'starknet';
+import { IStarknetWindowObject } from 'get-starknet';
+
+export class NoStarknetProviderError extends Error {
+  public constructor() {
+    super();
+    this.name = this.constructor.name;
+    this.message = 'No Starknet Provider was found on window object';
+  }
+}
+
+export class BraavosConnector extends AbstractConnector {
+  public async activate(): Promise<ConnectorUpdate> {
+    const wallet = this.getBraavosWallet();
+
+    if (!wallet) {
+      throw new NoStarknetProviderError();
+    }
+
+    wallet.on('accountsChanged', this.handleAccountsChanged);
+
+    let account: AccountInterface | undefined = wallet.account;
+    let connectedAddress: string | undefined = wallet.selectedAddress;
+
+    if (!account) {
+      [connectedAddress] = (await wallet.enable()) ?? [];
+      account = wallet.account;
+    }
+
+    return {
+      provider: wallet.provider,
+      chainId: 5,
+      ...(account ? { account } : {}),
+      ...(connectedAddress ? { connectedAddress } : {}),
+    };
+  }
+
+  private handleAccountsChanged = (accountAddresses: string[]) => {
```

Comment:
```suggestion
  private handleAccountsChanged (accountAddresses: string[]) {
```
I think we should use normal functions instead of arrow functions to maintain consistency?

Reply:
it's just a nicer way to avoid the -
```
this.handleAccountsChanged = this.handleAccountsChanged.bind(this);
```
binding in the (non-required) constructor, if it bothers you then I can return it of course.

Reply:
That makes sense. Thank you. Let's use arrow function then.